src/daemon/XorgDisplayServer.h

```diff
@@ -39,7 +39,7 @@ namespace SDDM {
 
         QString sessionType() const;
 
-        QString cookie() const;
+        const QByteArray cookie() const;
```

Comment:
This probably meant
```suggestion
        const QByteArray &cookie() const;
```

Reply:
There's little value in const& on implicitly shared objects. I'd suggest to just leave it as `QByteArray cookie() const;` as the const has also no value.

Reply:
I have little experience with Qt, and I didn't know about implicit sharing, thanks!
It obviously won't matter here, but I wonder what would be faster in a hot code section, a reference or a shallow copy? 🤔

By the way, this method is only used in [a single place](https://github.com/sddm/sddm/blob/b7155e6f7db9f34dc9e69c9f525110657bfddd9d/src/helper/UserSession.cpp#L60), and it always makes a copy (although a shallow one since implicit sharing). If my suggestion is the one that ends up being accepted, in this line a const& type should be used. I would also use `auto` there, like so:
```cpp
            const auto& cookie = qobject_cast<HelperApp*>(parent())->cookie();
```

Reply:
This is no hot section or anything, so it doesn't matter. It should just be QByteArray.